### PR TITLE
Stargate blog

### PR DESCRIPTION
**Pull request by Falydoor**

### CODE REVIEW

File: posts/stargate.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+- Theo Lebrun
+tags:
+- Cassandra
+- Big Data
+date: 2021-01-14T14:50:55.000Z
+title: "Store and query any data effortlessly using Stargate"
+image:
+---
+
+[Stargate](https://stargate.io/) is one of the latest shiny tools from [DataStax](https://www.datastax.com/) that will act as a data gateway to abtract Cassandra-specific concepts and remove barriers of entry for new software developers!
```
Comment (dsharpe) [inline posts/stargate.md:12]:
```suggestion
[Stargate](https://stargate.io/) is one of the latest shiny tools from [DataStax](https://www.datastax.com/) that will act as a data gateway to abstract Cassandra-specific concepts and remove barriers of entry for new software developers!
```

File: posts/stargate.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+Stargate is a full open source project ([GitHub repository](https://github.com/stargate/stargate)) that supports the most popular way of querying data (REST, GraphQL, Document and CQL API). By chosing their favorite way of querying data, developers will spend less time learning the API and more time on implementing new features.
```
Comment (dsharpe) [inline posts/stargate.md:14]:
```suggestion
Stargate is a full open source project ([GitHub repository](https://github.com/stargate/stargate)) that supports the most popular ways of querying data (REST, GraphQL, Document and CQL API). By choosing their favorite way of querying data, developers will spend less time learning the API and more time on implementing new features.
```

File: posts/stargate.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+The flag `DEVELOPER_MODE` tells Stargate to use its own Cassandra cluster but you can also use your own cluster. In order to provide your own cluster, you can follow the [official documentation](https://stargate.io/docs/stargate/1.0/developers-guide/install/install_existing_cstar.html).
+
+The Stargate container exposes 4 different services:
+- GraphQL interface for CRUD (**8080**)
+- REST authorization service for generating tokens (**8081**)
+- REST interface for CRUD (**8082**)
+- CQL service (**9042**)
+
+Make sure that the container was started without any errors before continuing this blog.
```
Comment (dsharpe) [inline posts/stargate.md:41]:
```suggestion
Make sure that the container was started without any errors before continuing.
```

File: posts/stargate.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+          "order": "ASC",
+          "column": "type"
+        }
+      ]
+    }
+  }
+}
+```
+
+The partition key is `manufacturer` and the clustering key is `type` so we can get all vehicles by manufacturer and sort them by type.
```
Comment (dsharpe) [inline posts/stargate.md:138]: Is that a "clustering key" or "sort key"?

File: posts/stargate.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+# Querying the data
+
+## GraphQL API
+
+Stargate will provide a GraphQL playground available at http://localhost:8080/playground, don't forget to set the header `x-cassandra-token` with the correct value at the bottom of the UI.
```
Comment (dsharpe) [inline posts/stargate.md:144]:
```suggestion
Stargate will provide a GraphQL playground available at http://localhost:8080/playground and don't forget to set the header `x-cassandra-token` with the correct value at the bottom of the UI.
```

File: posts/stargate.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    }
+  }
+}
+```
+
+The query should output something like below:
+
+![Output](https://raw.githubusercontent.com/Falydoor/blog-usa/stargate/images/2021/01/stargate-2.png)
+
+What makes GraphQL great is that you can define which fields to retrieve directly in your query to simplifies things a lot when querying complex APIs.
```
Comment (dsharpe) [inline posts/stargate.md:168]:
```suggestion
What makes GraphQL great is that you can define which fields to retrieve directly in your query to simplify things a lot when querying complex APIs.
```

File: posts/stargate.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+More details on how to use the GraphQL API can be found [here](https://stargate.io/docs/stargate/1.0/developers-guide/graphql-using.html).
+
+## REST API
+
+The query below can also be done using a more traditional way with the REST API and using predefined search terms.
```
Comment (dsharpe) [inline posts/stargate.md:174]:
```suggestion
The query below can also be done in a more traditional way with the REST API and using predefined search terms.
```

File: posts/stargate.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+The query output might be confusing because the document is flattened but a condition can be added to only retrieve the needed field.
+
+More details on how to use the Document API can be found [here](https://stargate.io/docs/stargate/1.0/developers-guide/document-using.html).
+
+## CQL API
+
+Finally, the CQL API will let you access the data using your favorite native language drivers like you would normally do with a regular Cassandra cluster.
+
+In our case, after being connected to our Stargate container following the [instructions here](https://stargate.io/docs/stargate/1.0/developers-guide/cql-using.html), we can run the following query to get all our vehicles:
```
Comment (dsharpe) [inline posts/stargate.md:236]:
```suggestion
In our case, after being connected to our Stargate container using the [instructions here](https://stargate.io/docs/stargate/1.0/developers-guide/cql-using.html), we can run the following query to get all our vehicles:
```

**Review by dsharpe — CHANGES_REQUESTED** [review]
Nice article!

File: posts/stargate.md (hunk 4 above)
Comment (Falydoor) [reply posts/stargate.md:138]: Clustering key yeah

**Review by dsharpe — APPROVED** [review]
LGTM

**Pull request merged**